### PR TITLE
feat: add 'md' size to sd-radio, sd-checkbox and sd-checkbox-group

### LINKED ISSUE #1326: fix: 🤔 consolidate sizes in input components (sd-radio/sd-checkbox/...)
## Current behavior
You cannot set `sd-radio-group` as size ="md" (options are 'lg', 'sm'). This means that even though `sd-radio-button` has a "md" size, there is no way to set this since `sd-radio-button` does not accept "md" as a size option and will reset all sizes set on the child `sd-radio-buttons` to the default "lg"

Image

## Expected behavior
You can set the `sd-radio-group`'s size as "md", in order to showcase medium sizes `sd-radio-buttons`

Image

## Steps to reproduce
Take a look at the Docs page of sd-radio-group and sd-radio-button, the group lacks a size "md" making sd-radio-buttons for this size incompatible.

## Technical Information
- [x] Design will rename the size in sd-radio / sd-checkbox & the group components
- [x] Code will add 3 sizes for sd-radio / sd-checkbox consistency wise

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### PR DESCRIPTION
**Pull request by Vahid1919**
## Description:

Closes #1326 

## Definition of Reviewable:

- [x] Documentation is created/updated
- [x] Migration Guide is created/updated
- [ ] E2E tests (features, a11y, bug fixes) are created/updated

- [x] Stories (features, a11y) are created/updated
- [x] relevant tickets are linked

### CODE REVIEW

File: packages/components/src/components/checkbox-group/checkbox-group.stories.ts [hunk 1]
```diff
@@ -47,6 +47,7 @@ export const Default = {
  * Use the `size` attribute to change the size:
```
Comment (karlbaumhauer) [inline packages/components/src/components/checkbox-group/checkbox-group.stories.ts:47]:
```suggestion
 * Use the `size` attribute to change the size (`md` equals `sm` but added for consistency reason on input elements):
```

File: packages/components/src/components/checkbox/checkbox.stories.ts [hunk 2]
```diff
@@ -39,6 +39,7 @@ export const Default = {
 /**
  * Use the `size` attribute to change the size of the input checkbox:
```
Comment (karlbaumhauer) [inline packages/components/src/components/checkbox/checkbox.stories.ts:40]:
```suggestion
 * Use the `size` attribute to change the size of the input checkbox (`md` equals `sm` but added for consistency reason on input elements):
```

File: packages/components/src/components/radio-button/radio-button.stories.ts
Comment (karlbaumhauer) [inline packages/components/src/components/radio-button/radio-button.stories.ts:1]: l50 should be changed as well in order to match the rest of the suggestions "(`md` equals `sm` but added for consistency reason on input elements)"

**Review by karlbaumhauer — CHANGES_REQUESTED** [review]
just added some suggestions in order to make the sm === md thing more clear.

**Comment by Vahid1919** [comment]
> I would also like to change the title as this is actually a feature of the components itself...?!?

Thanks for the feedback. According to figma and what I already understood, md should actually be equal to lg. So I will mention that in the docs. 

In regards to the title. I assumed it is a fix since it is a feature that was missing (hence broken radio-button-groups) and not a brand new feature that we are adding. What do you think @karlbaumhauer

**Comment by karlbaumhauer** [comment]
> > I would also like to change the title as this is actually a feature of the components itself...?!?
> 
> Thanks for the feedback. According to figma and what I already understood, md should actually be equal to lg. So I will mention that in the docs.
oh sh**t... mixed up the lines, sorry. I updated my suggestions.

> In regards to the title. I assumed it is a fix since it is a feature that was missing (hence broken radio-button-groups) and not a brand new feature that we are adding. What do you think @karlbaumhauer
You are right for the radio-group. This would be a fix.
But... As you are changing the other components as well and adding a new value to the components properties (without fixing a bug), it would be a feature. Even though it actually does not do anything feature-like (only adding it for consistency reasons... ;-)).

In theory, you would need to split the changes in one fix (radio-group) and another feature to checkbox and checkbox-group.

**Review by yoezlem — APPROVED** [review]

**Review by smfonseca — APPROVED** [review]

**Comment by Vahid1919** [comment]
@karlbaumhauer I updated the docs to include the documentation. I also changed the title to feat since that better explains all the changes

**Review by karlbaumhauer — APPROVED** [review]

**Pull request merged**